src/components/notification/utils.ts

```diff
@@ -0,0 +1,60 @@
+export function mapNotification(notification, channelName) {
+  const channelText = channelName ? `#${channelName}` : 'a channel';
+
+  if (notification.notificationType === 'chat_channel_mention') {
+    let senderName = displayName(notification.originUser?.profileSummary);
+
+    return {
+      id: notification.id,
+      createdAt: notification.createdAt,
+      body: `${senderName} mentioned you in ${channelText}`,
+      originatingName: senderName,
+      originatingImageUrl: notification.originUser?.profileSummary?.profileImage,
+      isUnread: notification.isUnread,
+    };
+  } else if (notification.notificationType === 'chat_channel_message_replied') {
+    let senderName = displayName(notification.originUser?.profileSummary);
+
+    return {
+      id: notification.id,
+      createdAt: notification.createdAt,
+      body: `${senderName} replied to you in ${channelText}`,
+      originatingName: senderName,
+      originatingImageUrl: notification.originUser?.profileSummary?.profileImage,
+      isUnread: notification.isUnread,
+    };
+  } else if (notification.notificationType === 'chat_dm_mention') {
+    let senderName = displayName(notification.originUser?.profileSummary);
+    return {
+      id: notification.id,
+      createdAt: notification.createdAt,
+      body: `${senderName} mentioned you in a conversation`,
+      originatingName: senderName,
+      originatingImageUrl: notification.originUser?.profileSummary?.profileImage,
+      isUnread: notification.isUnread,
+    };
+  } else if (notification.notificationType === 'chat_dm_message_replied') {
+    let senderName = displayName(notification.originUser?.profileSummary);
+    return {
+      id: notification.id,
+      createdAt: notification.createdAt,
+      body: `${senderName} replied to you in a conversation`,
+      originatingName: senderName,
+      originatingImageUrl: notification.originUser?.profileSummary?.profileImage,
+      isUnread: notification.isUnread,
+    };
+  }
+
+  return null;
+}
+
+function displayName(profileSummary) {
+  let displayName = [
+    profileSummary?.firstName,
+    profileSummary?.lastName,
+  ]
+    .filter((e) => e)
+    .join(' ');
+  displayName = displayName || 'Someone';
```

Comment:
What's the difference b/w using `||` & `??` ? 🤔 I tend to always use `??`

Reply:
The difference, I believe, is `??` checks for `null` or `undefined` and then falls back to the second one whereas `||` checks for the javascript "falsy":

Eg:
`0 ?? 16` would evaluate to 0
`0 || 16` would evaluate to 16

So, it's a matter of whether you want to consider falsy values as a reason to use the default or if you only want to know if there's a value at all.

Reply:
ah, okay. So i am assuming `'' ?? 'some text'` would also return `''`? Thanks for sharing!

Reply:
![image](https://user-images.githubusercontent.com/43770/233381284-ea548d37-abf0-4e5a-a5b0-229bb3e2ce92.png)

Correct!